specification/blueprint/resource-manager/Microsoft.Blueprint/preview/2018-11-01-preview/blueprintAssignment.json

```diff
@@ -363,6 +381,10 @@
           "type": "string",
           "description": "ID of the published version of a blueprint definition."
         },
+        "scope": {
```

Comment:
For the assigned scope of the assignment, 1) should it be read-only since the actual scope(scopeParameter) is in the path? 2) Do you think enum might work better since we have only two possible values (ref:TargetScope in BlueprintDefinition.json)?

Reply:
Yeah changing the name to something like `targetScope` would make much more sense, however, we may not be able to do that it in the current API version, because that is going to be a breaking change. Actually, the scope property property has been existing in our back-end for a long time even before MG-level assignment was implemented.

For assignment we don't have such a field for now, but that we can add it in the next API version along with the change to the scope property name (we may also want to change `targeScope` in BP definition to `targetScopeType`).

Reply:
@alex-frankel It would be great to get your thoughts on this. Filiz and I had a conversation offline yesterday, and we wondered if we can improve the description to make it less confusing with the existing scope parameter. However we struggled to reword it...

Reply:
Here's my pass at the description:

```
The target subscription ID for the blueprint assignment (format: '/subscriptions/{subscriptionId}'). This property is only valid for management group level assignments.
```

Reply:
Thanks for the revision. However, for subscription level assignments it's also valid and users can still include it if they want. It just need to match the other scope parameter in path. Actually it always shows up in the GET body but we didn't put it in the Swagger before.

Reply:
I see - is it a breaking change/new api version to change that behavior? If I mismatch those two scopes do I get an error? It seems like this will only serve to trip people up.

Reply:
Yes, that's a breaking change. You will get an error if the two scopes doesn't match for subscription level assignments. The behavior is sort of hidden, but it existing for a very long time...

Reply:
Actually that raises another question: should we also surface it in 2017-11-11-preview? Technically that's the right thing to do, but it will definitely cause trouble like you mentioned...

Reply:
oh, interesting.  why did that scope property exist in the first place? If this is not new, I'm not too concerned about it. Let's go with what you had originally to say it's only required for MG level assignments

Reply:
Looks like Haitao created the property, and it was designed for MG-level assignment. It is also used in a few other places mostly for validation.

Reply:
We had another offline discussion and decided to change the name in path to resourceScope.